.github/renovate.json5

```
@@ -156,6 +156,18 @@
         '/gardener/etcd-druid/',
       ],
     },
+    {
+      // Group helm updates in one PR.
+      groupName: 'helm',
+      matchDatasources: [
+        'github-releases',
+        'go',
+      ],
+      matchPackageNames: [
+        '/helm\\.sh/helm/v3/',
+        '/helm/helm/',
```

I think the proper regex should look like below (proper escpaing of `.` and `/`).
I like to use websites like [regex101.com](https://regex101.com) to validate regular expressions for Renovate (using the ECMAScript/JavaScript flavor).

See their test here:
* https://regex101.com/r/NNpgNK/1
* https://regex101.com/r/3DLxu8/1

```suggestion
        '/helm\.sh\/helm\/v3/',
        '/helm\/helm/',
```

Alternatively, we could simplify here since we're not making use of regex features anyway:
```suggestion
        'helm.sh/helm/v3',
        'helm/helm',
```
Not using the regex delimiters `//` makes Renovate check for exact string matches / glob pattern matching:
https://docs.renovatebot.com/string-pattern-matching/

---

https://github.com/gardener/gardener/pull/10749/ was doing it in the second method, but https://github.com/gardener/gardener/pull/10817/files changed it to,
```
    {
      // Group etcd-druid updates in one PR.
      groupName: 'etcd-druid',
      matchDatasources: [
        'github-releases',
        'go',
      ],
      matchPackageNames: [
        '/github\\.com/gardener/etcd-druid/',
        '/gardener/etcd-druid/',
      ],
    },
```
I was just following that.

---

Interesting! I wasn't aware that Renovate's automatic config migration adapts matching patterns as well 🤔 
I just tested them again in the JavaScript dev console and yours work :) Please disregard my comment.